carbonserver/carbonserver.go

```diff
@@ -624,6 +671,24 @@ func (listener *CarbonserverListener) updateFileList(dir string) {
 	}
 
 	metricsKnown := uint64(0)
+
+	// populate index for all the metric names in cache
+	// the iteration takes place only when cache-scan is enabled in conf
+	for fileName := range cacheMetricNames {
```

Comment:
Just a comment: this has a side effect of random order while ranging over maps. So keeping directories in cacheMetricNames is probably unnecessary for trie index.

Reply:
I am not much familiar with the existing trie index implementation and logic. Can you please explain a little more about this?

Reply:
For trie index, it always create a dir if not exists when inserting a metric. So if dirs didn't come in order anymore, they don't have to be cached.
It doesn't matter much, could just be a small memory savings when both trie and cache-scan are enabled.